### ISSUE TITLE
Split up platforms into targets and containers

### ISSUE THREAD

**Issue by rcannood** [viash-io/viash#195]
```yaml
functionality:
  name: foo
  requirements:
    n_proc: 10
    memory: 10G
    commands: [ sed ]
targets/artifacts/runners/executors/...:
  - type: executable
    default_backend: native
  - type: nextflow
    directives: foo
  - type: snakemake
backend/containers/images/environments/...:
  - type: docker
    image: foo
    setup: 
      - type: r
        packages: tidyverse
 - type: conda
   ...
```

What to call the targets and runtime environments, though?

* Targets / artifacts / runners / executors / ... ?
* Environments / Runtime environments / Compute environments / runners / backends / deployment / infrastructure

**Comment by rcannood** [comment]
`viash ns` would also need to have arguments for choosing which targets and backends to build

```bash
$ viash ns build --target_type 'executable|nextflow' --backend_type 'native|docker|conda'
```

**Comment by rcannood** [comment]
As noted by Hendrik, a first step for the proposed changes could be to define helper functions in e.g. Config to get  information pertaining the containers (e.g. `Config.getContainers(): List[Container]`) and the executors (e.g. `Config.getExecutors(): List[Executor]`).

**Comment by rcannood** [comment]
Our current implementation uses Container, but this is not a good term for being able to represent "native", "docker", and "conda", for instance. Is "Engine" a better name? Whichever name we choose will end up in several spots. 

* Viash config, e.g. `engine: [ { type: native }, { type: docker, ... } ]`
* Building an executable: `viash build --engines 'native|docker|conda' --executor executable -o bin config.vsh.yaml`
* Running an executable: `./foo ---engine docker`

**Comment by tverbeiren** [comment]
Some other options: `context`, `runtime`, `environment`.

**Comment by rcannood** [comment]
context, runtime, environment all sound good to me. Better than container, at any rate.

**Comment by rcannood** [comment]
This feature was implemented in #510 and will be included in Viash 0.9.0
